runtime/doc/lua.txt

```diff
@@ -1581,4 +1581,72 @@ uri_to_fname({uri})                                       *vim.uri_to_fname()*
                 Return: ~
                     Filename
 
+==============================================================================
+Lua module: xdiff                                                  *lua-xdiff*
+
+xdl_diff({a}, {b}, {opts})                                    *vim.xdl_diff()*
```

Comment:
Why not call this simply `vim.diff()` ? 

Also  "xdiff" is an implementation detail. Maybe worth mentioning once, but doesn't need to be in the help tags.
 Seems like the interface is generic enough, it is not inherently bound to a particular implementation.

Reply:
The original intention was to add a minimally-opinionated, maximally-transparent interface into xdiff.

> Also "xdiff" is an implementation detail.

It is and it isn't. Xdiff is the de facto standard diff library and has several implementations with pretty much identical API's, so I wanted this to be as close to that as possible. I'm ok if we want to do our own thing with this and just treat xdiff as pure implementation detail. Despite the interface being a close mapping to xdiff's, you're right in that it is quite generic.

Reply:
As this is Lua API and not (versioned) Neovim API, we have a bit more leeway to change the interface later (if, e.g., we want to add alternative diff methods).

In any case, I'd prefer `vim.xdiff` over `vim.xdl_diff` (possibly even over `vim.diff`, although I much prefer that over the latter).

Reply:
Renamed to vim.diff